### PR TITLE
Remove usage of Helper() interface from protostore

### PR DESCRIPTION
**Pull request by krishnamiriyala**
- Add TRACE() for debugging with additional logging
    - LOG_LEVEL=trace enables TRACE() and logging at Trace level
    - Exposed MsgToPersist/MsgToFilter for supporting multiple
    operations using single GetTransaction
    - Added unit test for Transaction with protoStore and dataStore objects
    - Minor fixes to the DOCUMENTATION.md (still WIP)
    - Add daily run to catch issues with race conditions in unit-tests

### CODE REVIEW

File: pkg/datastore/logger.go [hunk 1]
```diff
@@ -37,6 +37,13 @@ const (
 	SAAS_PERSISTENCE = "persistence"
 )
 
+var TRACE = func(format string, v ...any) {
+}
```
Comment (jeyhun) [inline pkg/datastore/logger.go:41]: Why is this needed? Why not just have `log.Trace()` function calls, which will take effect if the log level is set to _Trace_?

**Comment by jeyhun** [comment]
Not related to this PR, but I think that you can embed `*logrus.Entry` in `gormLogger` instead of naming the relevant field. I don't see reasons for that field to be named.
(Writing this here since GitHub won't let me add a comment to a line of code that was not modified in a PR).

File: pkg/datastore/helper.go [hunk 2]
```diff
@@ -117,6 +119,9 @@ func FromConfig(l *logrus.Entry, authorizer authorizer.Authorizer, cfg DBConfig)
 				if tx := db.gormDBMap[dbrole.MAIN].Exec(stmt); tx.Error != nil {
 					err = ErrExecutingSqlStmt.Wrap(tx.Error).WithValue(SQL_STMT, stmt)
 					db.logger.Errorln(err)
+					if strings.Contains(tx.Error.Error(), "duplicate key value") {
```
Comment (jeyhun) [inline pkg/datastore/helper.go:122]: Could you give an example of an error that would contain "duplicate key value"? Does that error happen when a Postgres user/role already exists?
Is there a way to cast the error to a relevant struct and check for one of its fields to determine if it's a "duplicate key value" error? Or, can `errors.Is()` be used to determine if the error is of the certain type?

File: pkg/datastore/database.go [hunk 3]
```diff
@@ -56,10 +57,12 @@ const (
...
-	authorizer  authorizer.Authorizer // Allows or cancels operations on the DB depending on user's org. and service roles
-	gormDBMap   map[dbrole.DbRole]*gorm.DB
-	logger      *logrus.Entry
-	initializer func(db *relationalDb, dbRole dbrole.DbRole) error
+	sync.RWMutex
+	authorizer         authorizer.Authorizer // Allows or cancels operations on the DB depending on user's org. and service roles
+	gormDBMap          map[dbrole.DbRole]*gorm.DB
+	logger             *logrus.Entry
+	initializer        func(db *relationalDb, dbRole dbrole.DbRole) error
+	triggerFuncCreated bool
```
Comment (jeyhun) [inline pkg/datastore/database.go:65]: I propose that the `check_and_update_revision` trigger function is created by default within `initializer`.  If no struct registered with DAL needs revisioning, no big deal, there will just be one extra function in the DB. We'll be able to get rid of `sync.RWMutex` and `triggerFuncCreated` if the trigger function is created in `initializer`.

File: pkg/protostore/protostore.go [hunk 4]
```diff
@@ -453,28 +452,24 @@ func (p ProtobufDataStore) DropTables(msgs ...proto.Message) error {
 }
 
 func (p ProtobufDataStore) MsgToPersist(ctx context.Context, id string, msg proto.Message, md Metadata) (ProtoStoreMsg, error) {
```
Comment (jeyhun) [inline pkg/protostore/protostore.go:454]: @krishnamiriyala, could you add comments describing what this method does?

File: pkg/protostore/protostore.go [hunk 5]
```diff
@@ -453,28 +452,24 @@ func (p ProtobufDataStore) DropTables(msgs ...proto.Message) error {
...
+	// Fill up the actual protobuf message and metadata from md
+	pMsg.Msg = bytes
+	pMsg.ParentId = md.ParentId
+	pMsg.Revision = md.Revision
+	return pMsg, nil
 }
 
-func (p ProtobufDataStore) MsgForFind(ctx context.Context, id string, msg proto.Message) (ProtoStoreMsg, error) {
-	orgId, err := p.ds.Helper().GetAuthorizer().GetOrgFromContext(ctx)
+func (p ProtobufDataStore) MsgToFilter(ctx context.Context, id string, msg proto.Message) (ProtoStoreMsg, error) {
```
Comment (jeyhun) [inline pkg/protostore/protostore.go:471]: @krishnamiriyala, could you add comments describing what this method does?

**Comment by jeyhun** [comment]
@krishnamiriyala, what do you think of adding a unit test that would demonstrate performing multiple operations (within a single transaction) on DB tables created using datastore and protostore?

File: pkg/datastore/logger.go (hunk 1 above)
Comment (krishnamiriyala) [reply pkg/datastore/logger.go:41]: TRACE can be used anywhere without the need to create a logger instance to be passed around and this is just to disable logging when TRACE is not enabled. check TRACE initialization below.

File: pkg/datastore/helper.go (hunk 2 above)
Comment (krishnamiriyala) [reply pkg/datastore/helper.go:122]: Look at the unit test failures, it's very rare but the initializer needs to be synchronized across multiple runs. I think it might be fine after the mutex lock

File: pkg/protostore/protostore.go (hunk 4 above)
Comment (krishnamiriyala) [reply pkg/protostore/protostore.go]: done

**Comment by krishnamiriyala** [comment]
@jeyhun resolved the comments

**Review by jeyhun — APPROVED** [review]
